From 4106d69a346281d016999bfe49700eba87fce085 Mon Sep 17 00:00:00 2001
From: Alessandro Fontana <alessandro.fontana@pm.me>
Date: Sat, 29 Aug 2026 15:57:50 +0000
Subject: [PATCH] arm64: dts: exynos7870: cluster clocks on cpu0/cpu4 (phase C
 plumbing, dtsi part)

Wire the cluster clocks into the CPU nodes, exynos850 style: cpu0 gets
cluster0_clk from cmu_cpucl0, cpu4 gets cluster1_clk from cmu_cpucl1
(the policy-leader CPUs; cpufreq-dt uses the policy CPU's clock).

v2: this patch now touches ONLY exynos7870.dtsi. The board dts
(exynos7870-gtaxlwifi.dts) is not part of the kernel source tree at
prepare time - it is an APKBUILD source= file copied in during build -
so its changes (buck1/vdd_cpu, the two single-OPP tables at 902.2 MHz /
762500 uV, cpu-supply + operating-points-v2 on all eight CPUs) ship as
a full replacement file instead: exynos7870-gtaxlwifi.dts, delivered
alongside this patch. Same content as v1, different packaging.
---
 arch/arm64/boot/dts/exynos/exynos7870.dtsi | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm64/boot/dts/exynos/exynos7870.dtsi b/arch/arm64/boot/dts/exynos/exynos7870.dtsi
index da6c64b010..b10c3ab045 100644
--- a/arch/arm64/boot/dts/exynos/exynos7870.dtsi
+++ b/arch/arm64/boot/dts/exynos/exynos7870.dtsi
@@ -68,6 +68,8 @@ cpu0: cpu@0 {
 			compatible = "arm,cortex-a53";
 			reg = <0x0>;
 			enable-method = "psci";
+			clocks = <&cmu_cpucl0 CLK_CLUSTER0_SCLK>;
+			clock-names = "cluster0_clk";
 		};
 
 		cpu1: cpu@1 {
@@ -96,6 +98,8 @@ cpu4: cpu@100 {
 			compatible = "arm,cortex-a53";
 			reg = <0x100>;
 			enable-method = "psci";
+			clocks = <&cmu_cpucl1 CLK_CLUSTER1_SCLK>;
+			clock-names = "cluster1_clk";
 		};
 
 		cpu5: cpu@101 {
-- 
2.47.3

